src/acl/Gadgets.cc

```diff
@@ -243,6 +250,40 @@ aclDeregister(ACL *acl)
     }
 }
 
+void
+Acl::MakeDiscoverableByName(ACL * const acl)
+{
+    const auto result = ByNameIndex_.try_emplace(acl->name, acl);
+    assert(result.second); // ban duplicates to simplify, detect more caller bugs
+}
+
+ACL *
+Acl::FindByName(const char * const name)
+{
+    const auto pos = ByNameIndex_.find(name);
+    if (pos != ByNameIndex_.end())
+        return pos->second;
+    return nullptr;
```

Comment:
can we simplify with a one-liner?
```suggestion
    return pos == ByNameIndex_.end() ? nullptr : pos->second;
```